Objective-C/CBLReplicator.mm

```diff
@@ -118,20 +118,24 @@ - (void) clearRepl {
 
 
 - (void) start {
-    if (_repl) {
-        CBLWarn(Sync, @"%@ has already started", self);
-        return;
-    }
-    
-    CBLLog(Sync, @"%@: Starting", self);
-    _retryCount = 0;
-    [self _start];
+    dispatch_async(_dispatchQueue, ^{
+        if (_repl) {
+            CBLWarn(Sync, @"%@ has already started", self);
+            return;
+        }
+        
+        CBLLog(Sync, @"%@: Starting", self);
+        _retryCount = 0;
+        _suspended = NO;
+        [self _start];
+    });
 }
 
 
 - (void) retry {
     if (_repl || _rawStatus.level != kC4Offline)
```

Comment:
This accesses `_repl` on the calling queue, which isn't thread-safe since it's also used on the `_dispatchQueue` (e.g. in -start.) There are probably other places where this happens.